### PR TITLE
[spec] fix dp-attn + spec + trtllm-mla

### PR DESCRIPTION
**Pull request by hnyls2002**
To completely fix the issue, it requires

- [ ] Move most of the metadata sync (in `prepare_mlp_sync_batch`) into `Scheduler.prepare_mlp_sync_batch`.
- [ ] Maybe deprecate the API `skip_attn_backend_init`.
- [ ] Refactor `ForwardMode` into more reliable status representation.

### CODE REVIEW

**Comment by Fridge003** [comment]
cc @rainj-me

File: python/sglang/srt/layers/attention/trtllm_mla_backend.py [hunk 1]
```diff
@@ -985,25 +983,6 @@ def forward_extend(
...
-                    # Dynamic allocation for non-CUDA graph mode
-                    padded_q = torch.zeros(
-                        bs,
-                        metadata.max_seq_len_q,
-                        layer.tp_q_head_num,
-                        layer.head_dim,
-                        dtype=q.dtype,
-                        device=q.device,
-                    )
-                q = self.pad_draft_extend_query(
```
Comment (rainj-me) [inline python/sglang/srt/layers/attention/trtllm_mla_backend.py:1004]: After removing the pad_draft_extend_query call, maybe we should already delete the relate code.

File: python/sglang/srt/layers/attention/trtllm_mla_backend.py [hunk 2]
```diff
@@ -1022,15 +1001,6 @@ def forward_extend(
                 bmm1_scale=bmm1_scale,
             )
 
-            # Reshape output directly without slicing
-
-            if forward_batch.forward_mode.is_draft_extend(include_v2=True):
-                raw_out = self.unpad_draft_extend_output(
```
Comment (rainj-me) [inline python/sglang/srt/layers/attention/trtllm_mla_backend.py:1028]: Same here.

File: python/sglang/srt/speculative/eagle_worker_v2.py [hunk 3]
```diff
@@ -512,7 +512,7 @@ def _draft_extend_for_decode(
             )
         else:
             draft_logits_output, _ = self.draft_runner.forward(
-                forward_batch, skip_attn_backend_init=True
+                forward_batch, skip_attn_backend_init=False
```
Comment (rainj-me) [inline python/sglang/srt/speculative/eagle_worker_v2.py:515]: skip attn backend lead to the flashinfer backend cudagraph capture failure.
